File: src/main/java/com/github/pozo/java/collections/list/Lists.java
package com.github.pozo.java.collections.list;

import java.util.Arrays;
import java.util.List;

public class Lists {
    public static void main(String[] args) {
        listOf_withIndex_onEach_filter();
    }

    public static void listOf_withIndex_forEach() {
        List<String> list = Arrays.asList("aaa", "bbb");

        list.forEach(System.out::println);
    }

    public static void listOf_withIndex_onEach_filter() {
        List<String> list = Arrays.asList("aaa", "bbb");

        list.stream()
                .map(it -> it.substring(2))
                .filter(it -> it.startsWith("a"))
                .forEach(System.out::println);
    }
}

File: src/main/kotlin/com/github/pozo/kotlin/collections/list/Lists.kt
package com.github.pozo.kotlin.collections.list

fun main(args: Array<String>) {
    listOf_withIndex_onEach_filter()
}

fun listOf_withIndex_forEach() {
    val list = mutableListOf("aaa", "bbb").withIndex()
    list.forEach({ v ->
        println(v)
    })/*.filter {
        entry ->
        entry.value.length < 3
    }*/
}

fun listOf_withIndex_onEach_filter() {
    val list = mutableListOf("aaa", "bbb").withIndex()

    list.map { it ->
        it.value.substring(2)
    }.filter { it ->
        it.startsWith("a")
    }.forEach({ it ->
        println(it)
    })
}
